src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -973,6 +969,44 @@ public override void DidReceiveChallenge (NSUrlSession session, NSUrlSessionTask
 				}
 			}
 
+			bool InvokeServerCertificateCustomValidationCallback (HttpRequestMessage request, SecTrust secTrust)
+			{
+				if (sessionHandler.ServerCertificateCustomValidationCallback is null)
+					return false;
+
+				var originalChain = secTrust.GetCertificateChain (); // TODO does this work for older iOS and mac versions?
+				var certificates = new X509Certificate2 [originalChain.Length];
+				for (int i = 0; i < originalChain.Length; ++i)
+					certificates [i] = originalChain [i].ToX509Certificate2 ();
+
+				X509Certificate2? certificate = certificates.Length > 0 ? certificates [0] : null;
+
+				// TODO is NSURLAuthenticationMethodServerTrust used for every request or only when there is a problem
```

Comment:
NSURLAuthenticationMethodServerTrust states that will be used when we are dealing with a server certificate, not a client one. This value is set for a scope, so AFAIK it won't be executing all the time and the OS will know if you trusted it or not in a previous call.

Reply:
Thanks for the explanation. The thing I was wondering was if it will be executed also when the server certificate is a valid one.